src/inverted_index.h

```diff
@@ -28,9 +28,17 @@ typedef struct {
   uint32_t numDocs;
 } InvertedIndex;
 
+struct indexReadCtx;
+
+typedef union {
```

Comment:
document what this thingie is and what it's used for - after all it's an API used from the outside...